### ISSUE TITLE
Doesn't include other "x-amz-*" header fields

### ISSUE THREAD

**Issue by robbiehanson** [badslug/Paw-AWSSignature4DynamicValue#4]
For example, try to do a [Put Object Copy](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html), which requires the "x-amz-copy-source" header.

Since this header isn't included in the signature, S3 returns the following error:

```
<Error>
  <Code>AccessDenied</Code>
  <Message>There were headers present in the request which were not signed</Message>
  <HeadersNotSigned>x-amz-copy-source</HeadersNotSigned>
  <RequestId>...</RequestId>
  <HostId>...</HostId>
</Error>
```

Very helpful utility by the way !

**Comment by badslug** [comment]
Hi,

I'm on vacation for a week but I'll look into it when I'm back. It seems like it should be simple to accommodate. If you want to submit a PR I can review and merge. :)

**Comment by badslug** [comment]
Hopefully fixed as part of PR #6
